python/GafferSceneUI/RenderPassEditor.py

```diff
@@ -832,10 +832,14 @@ def __updateTabs( self ) :
 
 			tabGroup = self.getPlug().node()["tabGroup"].getValue()
 
+			tabNames = []
 			for groupKey, sections in RenderPassEditor._RenderPassEditor__columnRegistry.items() :
 				if IECore.StringAlgo.match( tabGroup, groupKey ) :
-					for section in sections.keys() :
-						self._qtWidget().addTab( section )
+					tabNames.extend( sections.keys() )
+			# Deduplicate sections while preserving order in case the same
+			# section has been registered to multiple matching groupKeys.
+			for name in list( dict.fromkeys( tabNames ) ) :
```

Comment:
I was going to do some hand-wringing here about whether or not we should rely on the ordering of dicts, but I was very pleased to discover that although it was only an implementation detail in Python 3.6, it was enshrined in the language spec in 3.7. This means we can make `Metadata.registerNode()`take dicts, which will be way less error prone than the current lists...